src/components/charts/discoverChartDetail/ChartVersionSelectorModal.tsx

```diff
@@ -0,0 +1,209 @@
+        setSelectedChartValue(null)
+    }
+
+    const renderListHeader = (): JSX.Element => {
+        return (
+            <div className="flex content-start">
+                <button type="button" className="transparent pl-16 pr-16" onClick={togglePageState}>
+                    <Back className="icon-dim-20" />
+                </button>
+                <div>
+                    <div className="fw-6 fs-16 cn-9">{appStoreApplicationName}</div>
+                    <div className="fw-4 fs-13 cn-9">Select a {selectedValueType} value</div>
+                </div>
+            </div>
+        )
+    }
+    const renderValueList = (): JSX.Element => {
+        return (
+            <div style={{ height: 'calc(100vh - 170px)' }}>
+                <div className="chart-value-row fw-6 cn-7 fs-12 border-top border-bottom text-uppercase pt-8 pr-16 pb-8 pl-16">
+                    <div className="pr-16"></div>
+                    <div className="pr-16">Name</div>
+                    <div>Chart Version</div>
+                </div>
+                {(selectedValueType === 'preset' ? presetChartValueList : deployedChartValueList).map(
+                    (valueDetail, index) => (
+                        <div
+                            key={`chart-value-${index}`}
+                            className={`chart-value-row fw-4 cn-9 fs-13 pt-12 pr-16 pb-12 pl-16 ${
+                                selectedChartValue?.id === valueDetail.id ? 'active' : ''
+                            }`}
+                            onClick={() => {
+                                setSelectedChartValue(valueDetail)
+                            }}
+                        >
+                            <div className="pr-16">
+                                <File className="icon-dim-18 icon-n4 vertical-align-middle" />
+                            </div>
+                            <div className="pr-16">{valueDetail.name}</div>
+                            <div>{valueDetail.chartVersion}</div>
+                        </div>
+                    ),
+                )}
+            </div>
+        )
+    }
+    return (
+        <VisibleModal className="transition-effect">
+            <div
+                className={`modal__body mt-0 no-top-radius chart-version-selector p-0 ${
+                    isListpage ? 'no-bottom-radius' : ''
+                }`}
+            >
+                <div className={`header-container mt-20 mr-20 mb-20 ${isListpage ? '' : 'ml-20'}`}>
+                    {isListpage ? renderListHeader() : renderInitialHeader()}
+                    <button type="button" className="transparent" onClick={closePopup}>
+                        <Close className="icon-dim-20" />
+                    </button>
+                </div>
+                {isListpage ? renderValueList() : renderValueTypeList()}
+                {isListpage && (
+                    <div className="pt-20 pr-20 pb-20 pl-20 border-top right-align">
```

Comment:
Can isolate it as a functional component.

Reply:
done